apps/emqx_durable_storage/src/emqx_ds_lts.erl

```diff
@@ -0,0 +1,619 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_ds_lts).
```

Comment:
Thinking out loud: Might be worth adding later something like trimming, which clears out any edges except for `?PLUS`es from states with `?PLUS`es. Such trimmed tries can be migrated into new generations, maybe even automatically / periodically.